Plasma-based-cache/pom.xml

```diff
@@ -281,7 +281,7 @@
     <dependency>
       <groupId>com.intel.arrow</groupId>
       <artifactId>arrow-plasma</artifactId>
-      <version>0.17.0</version>
+      <version>4.0.0</version>
```

Comment:
I suggest we still use 0.17.0 here. Otherwise user need manually install arrow-plasma jar into their local maven repo. while 0.17.0 could be found in central maven repo. Any suggestion here? @zhixingheyi-tian